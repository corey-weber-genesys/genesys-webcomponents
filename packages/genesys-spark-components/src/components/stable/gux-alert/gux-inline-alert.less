@import (reference) '../../../style/color.less';
@import (reference) '../../../style/mixins.less';
@import (reference) '../../../style/spacing.less';
@import (reference) '../../../style/typography.less';
@import (reference) '../../../style/color-palette.less';

:host {
  display: block;
}

.gux-inline-alert {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  gap: @gux-spacing-xs;
  align-items: center;
  justify-content: flex-start;
  padding: @gux-spacing-2xs @gux-spacing-xs;
  border-radius: @gux-spacing-2xs;
  .body-font();

  gux-icon {
    flex-shrink: 0;
    align-self: flex-start;
    width: 16px;
    height: 16px;
    margin-top: @gux-spacing-3xs;
  }

  gux-tooltip-title {
    white-space: nowrap;
  }

  .gux-sr-only {
    .gux-sr-only-clip();
  }

  &.gux-info {
    color: @gux-black-80;
    background-color: @gux-grey-90;
    border: 1px solid @gux-grey-50;
  }

  &.gux-error {
    color: @gux-alert-red-30;
    background-color: @gux-alert-red-100;
    border: 1px solid @gux-alert-red-90;
  }

  &.gux-warning {
    color: @gux-alert-yellow-30;
    background-color: @gux-alert-yellow-100;
    border: 1px solid @gux-alert-yellow-90;
  }

  &.gux-success {
    color: @gux-alert-green-30;
    background-color: @gux-alert-green-100;
    border: 1px solid @gux-alert-green-90;
  }
}
