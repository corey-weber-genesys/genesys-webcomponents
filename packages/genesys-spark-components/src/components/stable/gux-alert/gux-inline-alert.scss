/**
  * @prop --gse-ui-alert-padding
  * @prop --gse-ui-alert-gap
  * @prop --gse-ui-alert-info-backgroundColor
  * @prop --gse-ui-alert-info-foregroundColor
  * @prop --gse-ui-alert-info-iconColor
  * @prop --gse-ui-alert-info-border-color
  * @prop --gse-ui-alert-info-border-width
  * @prop --gse-ui-alert-info-border-style
  * @prop --gse-ui-alert-success-backgroundColor
  * @prop --gse-ui-alert-success-foregroundColor
  * @prop --gse-ui-alert-success-iconColor
  * @prop --gse-ui-alert-success-border-color
  * @prop --gse-ui-alert-success-border-width
  * @prop --gse-ui-alert-success-border-style
  * @prop --gse-ui-alert-warning-backgroundColor
  * @prop --gse-ui-alert-warning-foregroundColor
  * @prop --gse-ui-alert-warning-iconColor
  * @prop --gse-ui-alert-warning-border-color
  * @prop --gse-ui-alert-warning-border-width
  * @prop --gse-ui-alert-warning-border-style
  * @prop --gse-ui-alert-error-backgroundColor
  * @prop --gse-ui-alert-error-foregroundColor
  * @prop --gse-ui-alert-error-iconColor
  * @prop --gse-ui-alert-error-border-color
  * @prop --gse-ui-alert-error-border-width
  * @prop --gse-ui-alert-error-border-style
  * @prop --gse-ui-alert-text-fontFamily
  * @prop --gse-ui-alert-text-lineHeight
  * @prop --gse-ui-alert-text-fontSize
  * @prop --gse-ui-alert-borderRadius
  */

@import '../../../style/mixins';

:host {
  display: block;
}

.gux-inline-alert {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  gap: var(--gse-ui-alert-gap);
  align-items: center;
  justify-content: flex-start;
  padding: var(--gse-ui-alert-padding);
  border: 1px solid;
  border-radius: var(--gse-ui-alert-borderRadius);

  gux-icon {
    flex-shrink: 0;
    align-self: flex-start;
    width: var(--gse-ui-icon-size-sm);
    height: var(--gse-ui-icon-size-sm);
    margin-top: 1px;
  }

  gux-tooltip-title {
    white-space: nowrap;
  }

  .gux-message-wrapper {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    gap: var(--gse-ui-alert-gap);
    font-family: var(--gse-ui-alert-text-fontFamily);
    font-size: var(--gse-ui-alert-text-fontSize);
    font-weight: 400;
    line-height: var(--gse-ui-alert-text-lineHeight);
  }

  .gux-sr-only {
    @include gux-sr-only-clip;
  }

  &.gux-info {
    color: var(--gse-ui-alert-info-foregroundColor);
    background-color: var(--gse-ui-alert-info-backgroundColor);
    border: var(--gse-ui-alert-info-border-width)
      var(--gse-ui-alert-info-border-style)
      var(--gse-ui-alert-info-border-color);
    gux-icon {
      color: var(--gse-ui-alert-info-iconColor);
    }
  }

  &.gux-error {
    color: var(--gse-ui-alert-error-foregroundColor);
    background-color: var(--gse-ui-alert-error-backgroundColor);
    border: var(--gse-ui-alert-error-border-width)
      var(--gse-ui-alert-error-border-style)
      var(--gse-ui-alert-error-border-color);
    gux-icon {
      color: var(--gse-ui-alert-error-iconColor);
    }
  }

  &.gux-warning {
    color: var(--gse-ui-alert-warning-foregroundColor);
    background-color: var(--gse-ui-alert-warning-backgroundColor);
    border: var(--gse-ui-alert-warning-border-width)
      var(--gse-ui-alert-warning-border-style)
      var(--gse-ui-alert-warning-border-color);
    gux-icon {
      color: var(--gse-ui-alert-warning-iconColor);
    }
  }

  &.gux-success {
    color: var(--gse-ui-alert-success-foregroundColor);
    background-color: var(--gse-ui-alert-success-backgroundColor);
    border: var(--gse-ui-alert-success-border-width)
      var(--gse-ui-alert-success-border-style)
      var(--gse-ui-alert-success-border-color);
    gux-icon {
      color: var(--gse-ui-alert-success-iconColor);
    }
  }
}
