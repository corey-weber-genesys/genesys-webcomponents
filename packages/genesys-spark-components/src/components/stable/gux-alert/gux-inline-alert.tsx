import { Component, Element, h, JSX, Prop } from '@stencil/core';
import { buildI18nForComponent, GetI18nValue } from '../../../i18n';
import { trackComponent } from '@utils/tracking/usage';
import { GuxAlertAccent } from './gux-inline-alert.types';
import translationResources from './i18n/en.json';

/**
 * @slot - Slot for message.
 */

@Component({
  styleUrl: 'gux-inline-alert.less',
  tag: 'gux-inline-alert',
  shadow: true
})
export class GuxAlert {
  private i18n: GetI18nValue;

  @Element()
  root: HTMLElement;

  @Prop()
  accent: GuxAlertAccent = 'info';

  private getIcon(accent: GuxAlertAccent): string {
    switch (accent) {
      case 'info':
        return 'alert-info';
      case 'success':
        return 'alert-success';
      case 'warning':
        return 'alert-warning-triangle';
      case 'error':
        return 'alert-warning-octogon';
      default:
        return 'alert-info';
    }
  }

  async componentWillLoad(): Promise<void> {
    trackComponent(this.root, { variant: this.accent });
    this.i18n = await buildI18nForComponent(this.root, translationResources);
  }

  render(): JSX.Element {
    return (
      <div
        class={{
          'gux-inline-alert': true,
          [`gux-${this.accent}`]: true
        }}
      >
        <gux-icon icon-name={this.getIcon(this.accent)} decorative></gux-icon>
        <div class="gux-message-wrapper">
          <div class="gux-sr-only">{this.i18n(this.accent)}</div>
          <slot />
        </div>
      </div>
    ) as JSX.Element;
  }
}
