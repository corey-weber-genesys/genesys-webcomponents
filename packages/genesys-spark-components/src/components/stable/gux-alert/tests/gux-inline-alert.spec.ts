import { newSpecPage } from '@stencil/core/testing';
import { GuxAlert } from '../gux-inline-alert';

const components = [GuxAlert];
const language = 'en';

describe('gux-alert-beta', () => {
  describe('#render', () => {
    [
      `<gux-alert-beta accent="info">Note: This is an information alert.</gux-alert-beta>`,
      ` <gux-alert-beta accent="success">Note: This is a success alert.</gux-alert-beta>`,
      `<gux-alert-beta accent="warning-octogon"
      >Note: This is a warning alert.</gux-alert-beta
    >`,
      `<gux-alert-beta accent="warning-triangle"
    >Note: This is a warning alert.</gux-alert-beta
  >`
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
