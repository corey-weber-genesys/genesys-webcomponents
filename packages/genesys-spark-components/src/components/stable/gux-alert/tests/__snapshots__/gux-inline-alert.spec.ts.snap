// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-alert-beta #render should render component as expected (1) 1`] = `
<gux-alert-beta
  accent="info"
>
  Note: This is an information alert.
</gux-alert-beta>
`;

exports[`gux-alert-beta #render should render component as expected (2) 1`] = `
<gux-alert-beta
  accent="success"
>
  Note: This is a success alert.
</gux-alert-beta>
`;

exports[`gux-alert-beta #render should render component as expected (3) 1`] = `
<gux-alert-beta
  accent="warning-octogon"
>
  Note: This is a warning alert.
</gux-alert-beta>
`;

exports[`gux-alert-beta #render should render component as expected (4) 1`] = `
<gux-alert-beta
  accent="warning-triangle"
>
  Note: This is a warning alert.
</gux-alert-beta>
`;
